include/zephyr/drivers/psi5/psi5.h

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Peripheral Sensor Interface (PSI5) driver API.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_PSI5_H_
+#define ZEPHYR_INCLUDE_DRIVERS_PSI5_H_
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief PSI5 Interface
+ * @defgroup psi5_interface PSI5 Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/**
+ * @brief PSI5 frame type
+ */
+enum psi5_frame_type {
+	PSI5_SERIAL_FRAME_4_BIT_ID,
+	PSI5_SERIAL_FRAME_8_BIT_ID,
+	PSI5_DATA_FRAME
+};
+
+/**
+ * @brief PSI5 frame structure
+ */
+struct psi5_frame {
+	enum psi5_frame_type type;
+
+	union {
+		uint32_t data;
+
+		struct {
+			uint8_t id;
+			uint16_t data;
+		} serial;
+	};
+
+	uint32_t timestamp;
+	uint8_t crc;
+	uint8_t slot_number;
+};
+
+/**
+ * @brief Defines the application callback handler function signature for sending.
+ *
+ * @param dev        Pointer to the device structure for the driver instance.
+ * @param channel    The hardware channel of the driver instance.
+ * @param status     PSI5 status.
+ * @param user_data  User data provided when the frame was sent.
+ */
+typedef void (*psi5_tx_callback_t)(const struct device *dev, uint8_t channel, int status,
```

Comment:
It seems like the definition for a possible value of `status` is missing.

Reply:
I mentioned the possible values in description